packages/mdc-select/foundation.ts

```diff
@@ -87,15 +93,41 @@ export class MDCSelectFoundation extends MDCFoundation<MDCSelectAdapter> {
     this.helperText_ = foundationMap.helperText;
   }
 
-  setSelectedIndex(index: number) {
-    this.adapter_.setSelectedIndex(index);
-    this.adapter_.closeMenu();
+  /** Returns the index of the currently selected menu item, or -1 if none. */
+  getSelectedIndex(): number {
+    return this.selectedIndex_;
+  }
+
+  setSelectedIndex(index: number, closeMenu = false) {
+    const previouslySelectedIndex = this.selectedIndex_;
+    this.selectedIndex_ = index;
+
+    this.adapter_.setSelectedText(
+        this.selectedIndex_ === numbers.UNSET_INDEX ?
+            '' : this.adapter_.getMenuItemTextAtIndex(this.selectedIndex_).trim());
+
+    if (previouslySelectedIndex !== numbers.UNSET_INDEX) {
+      this.adapter_.toggleClassAtIndex(previouslySelectedIndex, cssClasses.SELECTED_ITEM_CLASS, false);
+      // Remove the ARIA_SELECTED attribute.
+      this.adapter_.setAttributeAtIndex(previouslySelectedIndex, strings.ARIA_SELECTED_ATTR);
+    }
+    if (this.selectedIndex_ !== numbers.UNSET_INDEX) {
+      this.adapter_.toggleClassAtIndex(this.selectedIndex_, cssClasses.SELECTED_ITEM_CLASS, true);
+      this.adapter_.setAttributeAtIndex(this.selectedIndex_, strings.ARIA_SELECTED_ATTR);
+    }
+    this.layout();
+
+    if (closeMenu) {
+      this.adapter_.closeMenu();
+    }
+
     const didChange = true;
     this.handleChange(didChange);
   }
 
   setValue(value: string) {
-    this.adapter_.setValue(value);
+    const index = this.adapter_.getIndexOfMenuItemWithAttribute(strings.VALUE_ATTR, value);
```

Comment:
We're using DOM as source of truth here, we can store array of values in foundation instead where index serves as item key. This'll be retrieved on init for first time.

Example:
`this.adapter_.getItemValues(): string[]`